lib/identifiers/doi.rb

```diff
@@ -33,11 +33,21 @@ def self.extract(str)
         .to_s
         .downcase
         .scan(PATTERN)
-        .map { |doi|
-          next doi if doi =~ VALID_ENDING
+        .map { |doi| strip_punctuation(doi) }
+        .compact
+    end
+
+    def self.extract_one(str)
+      match = str.to_s.downcase[PATTERN]
```

Comment:
As we don't have private class methods in Ruby, it's possible someone will use this API to extract the first DOI from a search text. In this case, I wanted to make sure we go through the same normalisation process as `extract`.